### PR TITLE
Ingest cloudwaf entities

### PR DESCRIPTION
**Pull request by eXtremeX**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Summary

- The following new entities are **now** created:

| Resources | Entity `_type`    | Entity `_class` |
| --------- | ----------------- | --------------- |
| CloudWAF  | `sigsci_cloudwaf` | `Firewall`      |

- The following new relationships are **now** created:

| Source Entity `_type` | Relationship `_class` | Target Entity `_type` |
| --------------------- | --------------------- | --------------------- |
| `sigsci_corp`         | **HAS**               | `sigsci_cloudwaf`     |

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [X] This change requires a documentation update

### General Development Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [X] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [X] My changes properly paginate the target service provider's API
- [X] My changes properly handle rate limiting of the target service provider's
      API
- [X] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [X] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [X] I have updated the `CHANGELOG.md` file to describe my changes
- [X] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/steps/cloudwaf/converter.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        _type: Entities.CLOUD_WAF._type,
+        _class: Entities.CLOUD_WAF._class,
+        _key: getCloudWAFKey(cloudWAF.id),
+        name: cloudWAF.name,
+        displayName: cloudWAF.name,
+        category: ['application'],
+        description: cloudWAF.description,
+        region: cloudWAF.region,
+        tlsMinVersion: cloudWAF.tlsMinVersion,
+        siteName: cloudWAF.workspaceConfigs[0].siteName,
```
Comment (VDubber) [inline src/steps/cloudwaf/converter.ts]: What if there are multiples or zero workspaceConfigs?

File: src/steps/cloudwaf/converter.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        'deployment.Status': cloudWAF.deployment.status,
```
Comment (VDubber) [inline src/steps/cloudwaf/converter.ts:32]: Is there are reason to keep the `.`? Why not deploymentStatus?

File: src/steps/cloudwaf/converter.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        useUploadedCertificates: cloudWAF.useUploadedCertificates,
+        createdBy: cloudWAF.createdBy,
+        createdOn: parseTimePropertyValue(cloudWAF.created),
+        updatedBy: cloudWAF.updatedBy,
+      },
+    },
+  });
+}
+
+export function createOrganizationCloudwafRelationship(
```
Comment (VDubber) [inline src/steps/cloudwaf/converter.ts]: nit: createOrganizationCloudwafRelationship -> createOrganizationCloudWafRelationship

File: src/client.ts [hunk 4]
```diff
@@ -91,6 +92,21 @@ export class SignalSciencesAPIClient {
     }
   }
 
+  public async iterateCloudWAFInstances(
+    corpName: string,
+    iteratee: ResourceIteratee<SignalSciencesCloudWAF>,
+  ): Promise<void> {
+    const endpoint = this.buildEndpoint(corpName, '/cloudwafInstances');
+
+    const { data } = await this.fetch(endpoint, {
```
Comment (VDubber) [inline src/client.ts:101]: This does not appear to currently support pagination. Did you see anything in the responses to indicate that pagination was supported/needed?

**Comment by VDubber** [comment]
Looks great, a few minor comments. 
Please ping me when you are ready for another review.

File: src/steps/cloudwaf/converter.ts (hunk 2 above)
Comment (eXtremeX) [reply src/steps/cloudwaf/converter.ts:32]: This is something suggested by @austinkelleher long time ago, we had quite a few scenarios where response object contained nested objects and one idea was to keep the original response's structure by using `.` to designate the "nesting level".

The `deploymentStatus` could work just as well, I think it just comes down to preference and patterns (if there is any concrete one for cases such as this one - would be great if there is one).

File: src/steps/cloudwaf/converter.ts (hunk 1 above)
Comment (eXtremeX) [reply src/steps/cloudwaf/converter.ts]: Great callout, what if we have `siteNames` property instead and map all the workspaceConfigs' siteNames (It'll then be `string[]`) into it?

File: src/client.ts (hunk 4 above)
Comment (eXtremeX) [reply src/client.ts:101]: Nothing about pagination, it just returns the data, similar to the other SS endpoints.
[The documentation](https://docs.fastly.com/signalsciences/api/#_corps__corpName__cloudwafInstances_get) also doesn't say anything about pagination.

File: src/steps/cloudwaf/converter.ts (hunk 2 above)
Comment (eXtremeX) [reply src/steps/cloudwaf/converter.ts:32]: I haven't made any of these changes just yet, but could easily if we decide we want that pattern instead.

**Review by VDubber — APPROVED** [review]
Awesome! thanks

**Pull request merged**